sdks/java/core/src/main/java/org/apache/beam/sdk/io/range/OffsetRangeTracker.java

```diff
@@ -290,12 +262,16 @@ public synchronized String toString() {
    */
   @VisibleForTesting
   OffsetRangeTracker copy() {
-    OffsetRangeTracker res =
-        new OffsetRangeTracker(this.getStartPosition(), this.getStopPosition());
-    res.setOffsetOfLastSplitPoint(this.getOffsetOfLastSplitPoint());
-    res.setLastRecordStart(this.getLastRecordStart());
-    res.setDone(this.isDone());
-    res.setSplitPointsSeen(this.getSplitPointsSeen());
-    return res;
+    synchronized (this) {
+      OffsetRangeTracker res = null;
+      synchronized (res) {
```

Comment:
This is synchronizing on `null`, since `res` is initialized to `null` above - hence the test failures.